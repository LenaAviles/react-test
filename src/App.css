.App-logo {  
  height: 2.5rem;
  padding: .2rem;
}

.App-logo:hover {
  animation: App-logo-spin 1 .3s linear;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
