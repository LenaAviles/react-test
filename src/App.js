import React, { Component } from 'react';
// import logo from './logo.svg';
import './App.css';

import { QUESTIONS } from './fixtures';

// COMPONENTS
import AppHeader from './components/header';
import AppMenu from './components/menu';
import Navigation from './components/navigation';
import QuestionList from './components/QuestionList';

class App extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isMenuOpen: false
    }
    this.handleOpen = this.handleOpen.bind(this);
  }
  handleOpen(){
    this.setState({isMenuOpen: !this.state.isMenuOpen})
  }

  render() {
    return (
      <div>
        <AppHeader
          onLeftIconClick={() => this.setState({isMenuOpen: true})}
        />
        <Navigation />
        <AppMenu
          open={this.state.isMenuOpen}
          clickOut={(isMenuOpen) => this.setState({ isMenuOpen })}
          itemClicked = {this.handleOpen}
        /> 
        <QuestionList questions = {QUESTIONS}/>       
      </div>
    );
  }
}

export default App;
