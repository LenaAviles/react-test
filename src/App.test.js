import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';
import ShallowRenderer  from 'react-test-renderer/shallow';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MuiThemeProvider><App /></MuiThemeProvider>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer
    .create(<MuiThemeProvider><App /></MuiThemeProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders shallow correctly', () => {
  const renderer = new ShallowRenderer()
  const tree = renderer
    .render(<App />);
  expect(tree).toMatchSnapshot();
});

