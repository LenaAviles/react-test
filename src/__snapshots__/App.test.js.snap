// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders shallow correctly 1`] = `
<div>
  <AppHeader
    onLeftIconClick={[Function]}
  />
  <Navigation />
  <AppMenu
    clickOut={[Function]}
    itemClicked={[Function]}
    open={false}
  />
  <QuestionList
    questions={
      Array [
        Object {
          "anonymously": true,
          "category": "1",
          "comments": Array [],
          "date": 2018-04-11T18:43:09.194Z,
          "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Nam, rerum.",
          "id": 0,
          "test": "blood",
          "user": "Lena",
        },
        Object {
          "anonymously": true,
          "category": "1",
          "comments": Array [],
          "date": 2018-04-11T18:43:09.194Z,
          "description": "Lorem ipsum dolor sit amet.",
          "id": 1,
          "test": "blood",
          "user": "Lena",
        },
        Object {
          "anonymously": true,
          "category": "1",
          "comments": Array [],
          "date": 2018-04-11T18:43:09.194Z,
          "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Molestias non natus perferendis cum in aspernatur architecto corporis, quidem minima accusantium dicta dolorem corrupti quas labore?",
          "id": 2,
          "test": "blood",
          "user": "Lena",
        },
        Object {
          "anonymously": true,
          "category": "2",
          "comments": Array [],
          "date": 2018-04-11T18:43:09.194Z,
          "description": "Lorem ipsum dolor sit.",
          "id": 3,
          "test": "blood",
          "user": "Lena",
        },
        Object {
          "anonymously": true,
          "category": "2",
          "comments": Array [],
          "date": 2018-04-11T18:43:09.194Z,
          "description": "Lorem ipsum dolor sit amet consectetur adipisicing elit. Mollitia facilis qui, numquam rem nesciunt natus!",
          "id": 4,
          "test": "blood",
          "user": "Lena",
        },
        Object {
          "anonymously": true,
          "category": "2",
          "comments": Array [],
          "date": 2018-04-11T18:43:09.194Z,
          "description": "Lorem ipsum, dolor sit amet consectetur adipisicing elit. Dicta, deleniti sunt exercitationem hic, quisquam at sint nisi voluptatum minus rem cum! Magni saepe dolorum sed rem, in aliquid nihil, architecto exercitationem ut ratione temporibus quod quas? Molestias temporibus harum veniam!",
          "id": 5,
          "test": "blood",
          "user": "Lena",
        },
      ]
    }
  />
</div>
`;
